pkg/dao/rpm.go

```diff
@@ -284,25 +280,25 @@ func (r rpmDaoImpl) InsertForRepository(repoUuid string, pkgs []yum.Package) (in
 		Where("checksum in (?)", checksums).
 		Model(&models.Rpm{}).
 		Pluck("uuid", &rpmUuids).Error; err != nil {
-		return rowsAffected, fmt.Errorf("failed retrieving rpms.uuid for the package checksums: %w", err)
+		return 0, fmt.Errorf("failed retrieving rpms.uuid for the package checksums: %w", err)
 	}
 
 	// Delete Rpm and RepositoryRpm entries we don't need
 	if err = r.deleteUnneeded(repo, rpmUuids); err != nil {
-		return rowsAffected, fmt.Errorf("failed to deleteUnneeded: %w", err)
+		return 0, fmt.Errorf("failed to deleteUnneeded: %w", err)
 	}
 
-	//Add the RepositoryRpm entries we do need
+	// Add the RepositoryRpm entries we do need
 	associations := prepRepositoryRpms(repo, rpmUuids)
-	if err = r.db.Clauses(clause.OnConflict{
+	result := r.db.Clauses(clause.OnConflict{
 		Columns:   []clause.Column{{Name: "repository_uuid"}, {Name: "rpm_uuid"}},
 		DoNothing: true}).
-		Create(&associations).
-		Error; err != nil {
-		return rowsAffected, fmt.Errorf("failed to Create: %w", err)
+		Create(&associations)
+	if result.Error != nil {
+		return result.RowsAffected, fmt.Errorf("failed to Create: %w", result.Error)
 	}
 
-	return rowsAffected, err
+	return result.RowsAffected, err
```

Comment:
I changed the count being returned by InsertForRepository to the rows affected on the repositories_rpms table, so it reflects rpm changes per repository instead of rpm changes overall.